src/mustache_core.js

```diff
@@ -289,8 +289,13 @@ var core = {
 				res = evaluator();
 			}
 
-
-			return res == null ? "" : ""+res;
+			if (res == null) {
+				return "";
+			}
+			if (res instanceof DocumentFragment) {
```

Comment:
instead of checking instanceof, check `res.nodeType === 11` which is a document fragment.

Reply:
Updated!